platform/mk2/hal/timer_stm32/timer_device_stm32.c

```diff
@@ -48,6 +48,7 @@ static struct state {
         uint16_t period;
         uint16_t duty_cycle;
         uint16_t q_period_ticks;
+        uint16_t pulse_count;
```

Comment:
No good reason, I guess I thought I was saving memory or something.  I'll make the change and update the PR.